electrumx/lib/atomicals_blueprint_builder.py

```diff
@@ -325,8 +326,37 @@ def calculate_output_blueprint_fts(cls, tx, ft_atomicals, operations_found_at_in
       should_split_ft_atomicals = is_split_operation(operations_found_at_inputs)
       if should_split_ft_atomicals:
           return AtomicalsTransferBlueprintBuilder.color_ft_atomicals_split(ft_atomicals, operations_found_at_inputs, tx, is_split_activated)
+      
+      should_custom_colored_ft_atomicals = is_custom_colored_operation(operations_found_at_inputs)
+      if should_custom_colored_ft_atomicals:
+          return AtomicalsTransferBlueprintBuilder.custom_color_ft_atomicals(ft_atomicals, operations_found_at_inputs, tx)
       # Normal assignment in all cases including fall through of failure to provide a target exponent in the above resubstantiation
       return AtomicalsTransferBlueprintBuilder.color_ft_atomicals_regular(ft_atomicals, tx, sort_fifo, is_split_activated)
+  
+  @classmethod
+  def custom_color_ft_atomicals(cls, ft_atomicals, operations_found_at_inputs, tx):
+    output_colored_map = {}
+    fts_burned = {}
+    cleanly_assigned = True
+    for atomical_id, atomical_info in sorted(ft_atomicals.items()):
+      remaining_value = atomical_info.total_atomical_value
+      for out_idx, txout in enumerate(tx.outputs):
+        expected_output_index = str(out_idx)
+        compact_atomical_id = location_id_bytes_to_compact(atomical_id)
+        expected_value = operations_found_at_inputs["payload"][compact_atomical_id].get(expected_output_index, 0)
+        if expected_value <= 0 or remaining_value <= 0:
+            continue
+        if expected_value > txout.value:
+            expected_value = txout.value
```

Comment:
This results in a burn correct?

Reply:
If the value to be colored in the payload exceeds the actual Satoshis contained, only the Satoshis actually contained in this UTXO will be colored, and the burn amount will also be calculated correctly.

Reply:
LGTM